pkg/stream/consumer_base.go

```diff
@@ -83,11 +97,31 @@ func NewBaseConsumer(ctx context.Context, config cfg.Config, logger log.Logger,
 		Encoding: settings.Encoding,
 	})
 
+	retryHandler := NewRetryHandlerNoop()
+
+	// disable the retry handler for inputs which have a retry mechanism on its own
+	if retryAware, ok := input.(RetryableInput); ok {
+		settings.Retry.Enabled = settings.Retry.Enabled && !retryAware.HasRetry()
+	}
+
+	if settings.Retry.Enabled {
```

Comment:
Should the retry resources also show up on the metadata server?

Reply:
Yes, makes sense, will add that..